Homework13/Tests/WebTableTests.cs

```diff
@@ -0,0 +1,138 @@
+            var filteredResultRowsCountWithSearchedValue = GenericPages.WebTablePage.GetCountOfFilteredRows(GenericPages.WebTablePage.GetListOfNotEmptyRows(filteredRows), firstName1);
+
+            // Check that: 1) Each result row contains the first name from the first entry. 2) Number of rows with the first name before the search is the same as after the search.
+            Assert.Multiple(() =>
+            {
+                Assert.AreEqual(filteredResultRowsCountWithSearchedValue, resultRowsCountWithSearchedValue);
+                Assert.AreEqual(filteredRowsCountWithSearchedValue, resultRowsCountWithSearchedValue);
+            });
+        }
+
+        // Test3: Add new entry.
+        [Test]
+        public void AddNewEntry()
+        {
+            GenericPages.WebTablePage.ClickAddNewEntryButton();
+
+            // Fill in form inputs.
+            GenericPages.WebTablePage.FillInFirstNameInput(firstName);
+            GenericPages.WebTablePage.FillInLastNameInput(lastName);
+            GenericPages.WebTablePage.FillInEmailInput(email);
+            GenericPages.WebTablePage.FillInAgeInput(age);
+            GenericPages.WebTablePage.FillInSalaryInput(salary);
+            GenericPages.WebTablePage.FillInDepartmentInput(department);
+            GenericPages.WebTablePage.ClickSubmitButton();
+
+            // Get text of new entry and compare it with expected.
+            Assert.AreEqual(entryValue, GenericPages.WebTablePage.GetTextOfCollectionElements(GenericPages.WebTablePage.GetNewRow()));
+        }
+
+        // Test4: Edit first entry.
+        [Test]
+        public void EditFirstEntry()
+        {
+            GenericPages.WebTablePage.ClickEditEntryButton();
+
+            // Clear form inputs.
+            GenericPages.WebTablePage.ClearFirstNameInput();
+            GenericPages.WebTablePage.ClearLastNameInput();
+            GenericPages.WebTablePage.ClearEmailInput();
+            GenericPages.WebTablePage.ClearAgeInput();
+            GenericPages.WebTablePage.ClearSalaryInput();
+            GenericPages.WebTablePage.ClearDepartmentInput();
+
+            // Fill in form inputs.
+            GenericPages.WebTablePage.FillInFirstNameInput(firstName);
+            GenericPages.WebTablePage.FillInLastNameInput(lastName);
+            GenericPages.WebTablePage.FillInEmailInput(email);
+            GenericPages.WebTablePage.FillInAgeInput(age);
+            GenericPages.WebTablePage.FillInSalaryInput(salary);
+            GenericPages.WebTablePage.FillInDepartmentInput(department);
+            GenericPages.WebTablePage.ClickSubmitButton();
+
+            // Get text of updated entry and compare it with expected.
+            Assert.AreEqual(entryValue, GenericPages.WebTablePage.GetTextOfCollectionElements(GenericPages.WebTablePage.GetEditedRow()));
+        }
+
+        // Test5: Delete first entry.
+        [Test]
+        public void DeleteFirstEntry()
```

Comment:
Bad approach of the test with current methods - it will fail if the first and second rows names have the same name

Reply:
discussed in direct messages, won't fix